seahorn/lib/utils.c

```diff
@@ -106,53 +106,68 @@ void assert_byte_cursor_equivalence(
 #define MAX_STRING_LEN MAX_BUFFER_SIZE
 #endif
 
-size_t sea_strlen(const char *str, size_t max) {
-  size_t i;
-  i = nd_size_t();
-  assume(i < max && max <= MAX_STRING_LEN);
-  assume(str[i] == '\0');
+// A strlen implementation
+size_t sea_strlen(const char *str) {
+  size_t len = 0;
+  for (size_t i = 0; i < MAX_STRING_LEN; i++) {
+    if (str[i] == '\0') {
+      break;
+    }
+    len++;
+  }
+  return len;
+}
+
+// This takes in a ptr to an allocated piece of memory with the desired
+// string len and initializes the string.
+void sea_init_str(const char *str, size_t str_len) {
```

Comment:
our sea_strlen is now lost but it was / will be useful when we only care about finding a zero terminator. Please restore it under some name

Reply:
I don't understand this comment. Earlier `sea_strlen` used to choose a length and create a valid string of that length.

The current implementation (with the changes you propose) should be able to find a zero terminator and give the location from start of the string.

Reply:
previous implementation without a loop was still useful approximation of strlen. We should not loose it. Just rename and document the difference

Reply:
done